src/asm2wasm.h

```diff
@@ -1100,10 +1102,25 @@ void Asm2WasmBuilder::processAsm(Ref ast) {
           auto& segment = wasm.table.segments[0];
           functionTableStarts[name] = segment.data.size(); // this table starts here
           Ref contents = value[1];
-          for (unsigned k = 0; k < contents->size(); k++) {
-            IString curr = contents[k]->getIString();
-            segment.data.push_back(curr);
-          }
+          if (compactFunctionTable) {
+            if (segment.data.size() == 0 && contents->size() > 0) {
+			  IString curr = contents[0]->getIString();
+			  segment.data.push_back(curr); //null fnptr
```

Comment:
How about if the contents are not empty, then note the first item as the null function. If the segment is empty, add it. And the null function can be used later down: instead of checking for `_` prefixing, we can compare to the null function. Also we can emit the null function instead of `Name()` for filling in the gaps (it must be a valid function).